session/txn.go

```diff
@@ -228,9 +259,36 @@ func (txn *LazyTxn) changeToInvalid() {
 	txn.stagingHandle = kv.InvalidStagingHandle
 	txn.Transaction = nil
 	txn.txnFuture = nil
-	atomic.StoreUint64(&txn.infoStartTS, 0)
-	atomic.StoreUint64(&txn.EntriesCount, 0)
-	atomic.StoreUint64(&txn.EntriesSize, 0)
+
+	txn.recreateTxnInfo(
+		0,
+		txninfo.TxnRunningNormal,
+		0,
+		0,
+		"",
+		nil)
+}
+
+func (txn *LazyTxn) onStmtStart(currentSQLDigest string) {
```

Comment:
Some benchmark like sysbench point select is very sensitive to main path changes, maybe we need to verify the possible cost doing begin/end for each statement?

Reply:
| With this PR | Workload | Threads | Table size | QPS | .95 lat |
|----|--|--|--|--|--|
| Y | oltp_point_select | 32  | 1,000,000 | 37558.47  | 1.32 |
| Y | oltp_point_select | 64 | 1,000,000 | 52974.07 | 2.61 |
| Y | oltp_point_select | 128 | 1,000,000 | 63404.03 | 5.09 |
|   | oltp_point_select | 32 | 1,000,000 | 37447.62 | 1.30 |
|   | oltp_point_select | 64 | 1,000,000 | 52580.28 | 2.66 |
|   | oltp_point_select | 128 | 1,000,000 | 63506.47 | 4.91 |

I did a brief test for point select and it looks no much difference.